### PR TITLE
Rely on request.Context() cancellation

### PR DESCRIPTION
**Pull request by cpuguy83**
The cancellable handler is no longer needed as the context that is
passed with the http request will be cancelled just like the close
notifier was doing.

Also updates gorilla/mux... This fixes an issue with mux usage of context for storing vars... took me a bit to figure out that calling `mux.Vars(r)` after I added my own context to `r` was messing everything up.
Also the old version is 2 years old.

### CODE REVIEW

**Review by thaJeztah — APPROVED** [review]
SGTM

**Comment by thaJeztah** [comment]
Hm failure; could this be due to a change in behavior in gorilla/mux? I recall we ran into this at some point, perhaps another attempt to update it

```
22:48:15 FAIL: docker_api_test.go:91: DockerSuite.TestAPIErrorNotFoundJSON
22:48:15 
22:48:15 docker_api_test.go:95:
22:48:15     c.Assert(httpResp.StatusCode, checker.Equals, http.StatusNotFound)
22:48:15 ... obtained int = 405
22:48:15 ... expected int = 404
22:48:15 
22:48:15

**Comment by thaJeztah** [comment]
could've been this one https://github.com/moby/moby/pull/30933 🤔 but think I had another one in mind

**Comment by thaJeztah** [comment]
Ah! This was the PR https://github.com/moby/moby/pull/37522#issuecomment-407844069

Same failure

**Comment by cpuguy83** [comment]
Damnit.

**Comment by cpuguy83** [comment]
Fixed 🤞

**Comment by thaJeztah** [comment]
I think I looked into that at the time with @vdemeester and it was because non-existing endpoints were still handled somewhere in our code, but it's been a while 🤔

File: api/server/server.go [hunk 1]
```diff
@@ -191,6 +192,7 @@ func (s *Server) createMux() *mux.Router {
 	notFoundHandler := httputils.MakeErrorHandler(pageNotFoundError{})
 	m.HandleFunc(versionMatcher+"/{path:.*}", notFoundHandler)
 	m.NotFoundHandler = notFoundHandler
+	m.MethodNotAllowedHandler = notFoundHandler
```
Comment (cpuguy83) [inline api/server/server.go]: gorilla/mux now uses a separate handler for non-existent routes. This changes it back to the old behavior.

Comment (thaJeztah) [reply api/server/server.go]: Ah! nice find; perhaps worth adding a comment, explaining why we do this?

Comment (thaJeztah) [reply api/server/server.go]: okz

**Review by thaJeztah — APPROVED** [review]
still LGTM (if green)

**Review by vdemeester — APPROVED** [review]
LGTM if green :green_apple:

**Comment by thaJeztah** [comment]
It's green (Ignoring codecov/patch) 🎉

**Pull request merged**